mapreader/loader/images.py

```diff
@@ -249,119 +267,189 @@ def add_shape(self, tree_level="parent"):
+
+        Parameters
+        ----------
+        tree_level : str, optional
+            Tree level, choices of "parent" or "child, by default "child"
+        verbose : bool, optional
+            If True, print verbose outputs, by default False
+        """
         list_items = list(self.images[tree_level].keys())
+        
         print(f"[INFO] Add center coordinates, tree level: {tree_level}")
+        
         for one_item in list_items:
-            self.add_center_coord_id(image_id=one_item, tree_level=tree_level)
+            
+            if tree_level == "parent":
+                if not "coord" in self.images[tree_level][one_item].keys():
+                    if verbose:
+                        print(f"'coord' could not be found in {one_item}.Suggestion: run add_metadata or addGeoInfo")
+                    continue
+            
+            if tree_level == "child":
+                par_id = self.images[tree_level][one_item]["parent_id"]
+                
+                if not "coord" in self.images["parent"][par_id].keys():
+                    if verbose:
+                        print(f"'coord' could not be found in {one_item}.Suggestion: run add_metadata or addGeoInfo")
+                    continue
+            
+            self.add_center_coord_id(image_id=one_item, tree_level=tree_level, verbose=verbose)
 
     def add_shape_id(self, image_id, tree_level="parent"):
-        """Add an image/array shape to self.images[tree_level][image_id]
+        """Add an image/array shape to image
 
         Parameters
         ----------
         image_id : str
-            image ID
+            Image ID
         tree_level : str, optional
-            Tree level, choices between parent and child (default: {"child"})
+            Tree level, choices of "parent" or "child, by default "parent"
         """
         myimg = mpimg.imread(self.images[tree_level][image_id]["image_path"])
+        #shape = (hwc)
         myimg_shape = myimg.shape
         self.images[tree_level][image_id]["shape"] = myimg_shape
 
-    def add_coord_increments_id(self, image_id, tree_level="parent"):
-        """Add pixel-wise dlon and dlat to self.images[tree_level][image_id]
+    def add_coord_increments_id(self, image_id, tree_level="parent", verbose=False):
+        """Add pixel-wise dlon and dlat to image
 
         Parameters
         ----------
         image_id : str
-            image ID
+            Image ID
         tree_level : str, optional
-            Tree level, choices between parent and child (default: {"child"})
+            Tree level, choices between "parent" or "child, by default "parent"
+        verbose : bool, optional
+            If True, print verbose outputs, by default False
         """
+
         if not "shape" in self.images[tree_level][image_id].keys():
             self.add_shape(tree_level=tree_level)
 
         if not "coord" in self.images[tree_level][image_id].keys():
-            raise ValueError(
```

Comment:
I agree that it would be clearer if the `add_shape_id`, `add_coord_increments_id` and `add_center_coord_id` functions were private. 

This is an appropriate workaround in lieu of #27.